### PR TITLE
Final internals

### PR DESCRIPTION
**Pull request by antonypap**
Actually winning the game :')

### CODE REVIEW

**Comment by antonypap** [comment]
LGTM

File: pkg/infra/teams/team3/fight_strat.go [hunk 1]
```diff
@@ -160,61 +170,43 @@ func (a *AgentThree) HandleFightProposal(m message.Proposal[decision.FightAction
...
+		Delta2ST := groupAvST - TSNavST
+		Delta2ATT := groupAvATT - TSNavATT
+		Delta2DEF := groupAvDEF - TSNavDEF
+
+		HPThreshold = myHP + alpha*Delta1HP + beta*Delta2HP
+		StaminaThreshold = myST + alpha*Delta1ST + beta*Delta2ST
+		AttackThreshold = myATT + alpha*Delta1ATT + beta*Delta2ATT
+		DefenseThreshold = myDEF + alpha*Delta1DEF + beta*Delta2DEF
+
+		return HPThreshold, StaminaThreshold, AttackThreshold, DefenseThreshold
```
Comment (MattSScott) [inline pkg/infra/teams/team3/fight_strat.go]: worth wrapping this in a `threshold struct` for clarity (multivariate returns are a bit gross)

File: pkg/infra/teams/team3/helpers.go [hunk 2]
```diff
@@ -156,3 +204,31 @@ func minMax4(isMax bool, nums [4]float64) float64 {
...
 	return ans
 }
+
+func (a *AgentThree) getGroupAvStats(baseAgent agent.BaseAgent) (float64, float64, float64, float64) {
+	groupAvHP := AverageArray(GetHealthAllAgents(baseAgent))
+	groupAvST := AverageArray(GetStaminaAllAgents(baseAgent))
+	groupAvATT := AverageArray(GetAttackAllAgents(baseAgent))
+	groupAvDEF := AverageArray(GetDefenceAllAgents(baseAgent))
+
+	return groupAvHP, groupAvST, groupAvATT, groupAvDEF
```
Comment (MattSScott) [inline pkg/infra/teams/team3/helpers.go]: similar to before - worth wrapping in `groupAv struct`

Comment (MattSScott) [reply pkg/infra/teams/team3/helpers.go]: Or, indeed, abstracting to a `avg struct` for all of these functions

**Review by MattSScott — COMMENTED** [review]
Reads very nicely. Only (not absolutely necessary) criticism would be to create a struct to wrap the multivariate returns

File: pkg/infra/teams/team3/fight_strat.go (hunk 1 above)
Comment (BusterBlack) [reply pkg/infra/teams/team3/fight_strat.go]: yep ok

File: pkg/infra/teams/team3/helpers.go (hunk 2 above)
Comment (BusterBlack) [reply pkg/infra/teams/team3/helpers.go]: created an avgerageStats struct and a ThresholdVal struct for the two multi-var returns

**Comment by MattSScott** [comment]
LGTM!

**Pull request merged**